### ISSUE TITLE
Instructions for upgrading Docker fail with 'Unable to locate package docker-ce'

### ISSUE THREAD

**Issue by edmorley** [travis-ci/docs-travis-ci-com#1258]
The Travis Docker docs explain how to update the pre-installed Docker version to a newer version, here:
https://docs.travis-ci.com/user/docker/#Installing-a-newer-Docker-version

Using those steps like so:
```yaml
dist: trusty
sudo: required
services:
  - docker
before_install:
  - sudo apt-get update
  - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
```

...results in:
```
$ sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package docker-ce
```

(See: https://travis-ci.org/mozilla/treeherder/builds/242782924#L408)

**Comment by plaindocs** [comment]
Hi @edmorley - thanksffor reporting.

I think that was fixed in #1256 , could you take another look at the live docs and let me know?

**Comment by edmorley** [comment]
Many thanks for the fast reply! 

I must have opened the docs page just before that landed.

Looking at the Travis log above, I see APT installing from https://apt.dockerproject.org/repo/ - so I don't think it's that it's using the default Ubuntu docker package (ie: https://packages.ubuntu.com/trusty/docker), but that's it's using the old-style upstream repo.

Perhaps the Travis images should be updated to switch from apt.dockerproject.org to download.docker.com?

**Comment by plaindocs** [comment]
@cotsog ?

**Comment by cotsog** [comment]
@edmorley: thanks for the additional details! You are right that `docker-engine` isn't downloaded from the default Ubuntu APT repo. I'll change the documentation to reflect this.

I think we will indeed want to switch to download.docker.com in the near future but we cannot give you an ETA at this time.

**Comment by cotsog** [comment]
@edmorley @plaindocs: See https://github.com/travis-ci/docs-travis-ci-com/pull/1259

**Comment by edmorley** [comment]
Reading further I've found:
* apt.dockerproject.org is deprecated and people are encouraged to use download.docker.com instead (https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository + see https://github.com/moby/moby/issues/31514#issuecomment-283995789)
* the new repo calls the package `docker-ce` rather than `docker-engine`
* a PR to switch Travis to the new repo has already landed (travis-ci/travis-cookbooks#843), however presumably isn't deployed yet

If that PR isn't going to be deployed soon, perhaps we should revert the docs back to how they were before #1248 / #1256?

Once the PR is deployed the docs can then just reference `docker-ce` only.

**Comment by plaindocs** [comment]
Thanks for your help tracking this down @edmorley and @cotsog for all the info.

Looks like the PR goes live in about a week, so as long as apt.dockerproject is functional, I'd be tempted to merge in #1259 , and then tweak it again in a week instead of reverting. 

What do you think?

**Comment by edmorley** [comment]
A week is shorter than I expected, so that plan sounds great :-)

**Comment by cotsog** [comment]
@edmorley @plaindocs: SGTM 👍

**Comment by edmorley** [comment]
The last piece of this was fixed in #1392.

**Comment by Godley** [comment]
I just got stung by this from trying to run the old instructions - suspect dockerproject is now deprecated? Is there a reason why both instruction sets are on there if one of them doesn't work?
